apps/api/views/homework.py

```diff
@@ -32,11 +36,25 @@ class SubmissionViewSet(ModelViewSet):
     serializer_class = SubmissionSerializer
     permission_classes = (SubmissionPermissionCheck,)
 
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+
     def perform_create(self, serializer):
         new_sub = serializer.save()
         if new_sub.pk and not new_sub.submitted_to_challenge:
-            if not new_sub.definition.questions_only:
-                post_submission(new_sub.pk)
+            if not new_sub.definition.questions_only and new_sub.github_url:
+                logger.info("Submission is a github submission!")
+                post_submission.delay(new_sub.pk)
+            elif self.request.data['file']:
+                logger.info("Submission is a direct upload!")
+                new_sub.is_direct_upload = True
+                new_sub.save()
+                # We can't serialize TemporaryUploadedFiles for celery
```

Comment:
what does this comment mean? Can we remove it if we're not doing `.apply_async` or something?

Reply:
It was just the explanation for why we're not doing async there. Can just pull it if it doesn't make sense.